### DISCUSSION TITLE
Running profiling tool without Python

### DISCUSSION THREAD

**Discussion by havefuncoding1** [iovisor/bcc#4780] [Q&A]
Hi, I am trying to profile CPU usage. [This](https://github.com/iovisor/bcc/blob/master/man/man8/profile.8) document covers how to use the tool, and I believe this tool is actually Python script [here](https://github.com/iovisor/bcc/blob/master/tools/profile.py). Trouble is my environment is quite strict and I can't install Python package. Is there a native C/C++ version of the profiler available? Not sure if I am on the right track, but I did find a profile.bpf.c [here](https://github.com/libbpf/bpftool/blob/6eb3e20583da834da18ea3011dcefd08b3493f8d/src/skeleton/profiler.bpf.c). Is this something I need to leverage? Thanks in advance for your help.

**Comment by Anyel-ec** [comment]
Hello my bro. 

I understand your situation. If your environment does not allow installing Python packages, considering using a native C/C++ profiler is a good idea. The "profile.bpf.c" file you found seems to be a promising option.

You must compile and run that C code to get CPU usage information. Make sure you understand how it works and adapt it to your needs. It may be an effective solution for your case.

Good luck! 🚀

**Reply by havefuncoding1** [reply]
Thanks a lot, @Anyel-ec

**Comment by ekyooo** [comment]
Hello.

You can get the C version of Profile from this PR.
https://github.com/iovisor/bcc/pull/3782

This is a converted version of the BCC Python profile.
If you encounter any issues while using it, please contact me. 
Thank you.
